syntax = "proto3";

// Interface de serviço
service ConnectionMiddleware {
  rpc Comunication (Request) returns (Reply) {}
}

// Mesagem de requisição enviada pelo cliente
message Request {
  enum TypeReq {
    ADDMATRICULA = 0;
    PUTNOTA = 1;
    PUTFALTAS = 2;
    GETALUNOS = 3;
    GETBOLETIM = 4;
    EXIT = 5;
  }

  TypeReq option = 1;
  Matricula matricula = 2;
}
// Menssagem de resposta enviada pelo servidor
message Reply {
  string message = 1;
  repeated Aluno student = 2;
  repeated DiscBoletimAluno report_card = 3;
}

// Menssagem especifica de um discplina do boletim de um aluno
message DiscBoletimAluno {
  string nome_disc = 1;
  float nota = 2;
  int32 faltas = 3;

}
   
// Messagens referentes a entidades do banco de dados
message Curso {
  int32 codigo = 1;
  string nome = 2;
}

message Disciplina {
  string codigo = 1;
  string nome = 2;
  string professor = 3;
  int32 cod_curso = 4;
}

message Matricula {
  int32 RA = 1;
  string cod_disciplina = 2;
  int32 ano = 3;
  int32 semestre = 4;
  float nota = 5;
  int32 faltas = 6;
}

message Aluno {
  int32 RA = 1;
  string nome = 2;
  int32 periodo = 3;
  int32 cod_curso = 4;
}